src/app/georeferencer/qgsimagewarper.h

```diff
@@ -23,26 +23,42 @@
 #include "qgspointxy.h"
 #include "qgscoordinatereferencesystem.h"
 #include "qgsogrutils.h"
+#include "qgstaskmanager.h"
 
 class QgsGeorefTransform;
-class QProgressDialog;
-class QWidget;
+class QgsFeedback;
 
 class QgsImageWarper
 {
     Q_DECLARE_TR_FUNCTIONS( QgsImageWarper )
+    Q_GADGET
 
   public:
-    explicit QgsImageWarper( QWidget *parent );
 
-    enum ResamplingMethod
+    QgsImageWarper();
+
+    enum class ResamplingMethod : int
     {
       NearestNeighbour = GRA_NearestNeighbour,
-      Bilinear         = GRA_Bilinear,
-      Cubic            = GRA_Cubic,
-      CubicSpline      = GRA_CubicSpline,
-      Lanczos          = GRA_Lanczos
+      Bilinear = GRA_Bilinear,
+      Cubic = GRA_Cubic,
+      CubicSpline = GRA_CubicSpline,
+      Lanczos = GRA_Lanczos
+    };
+    Q_ENUM( ResamplingMethod )
+
+    //! Task results
+    enum class Result
+    {
+      Success, //!< Warping completed successfully
+      Canceled, //!< Task was canceled before completion
+      InvalidParameters, //!< Invalid transform parameters
+      SourceError, //!< Error reading source
+      TransformError, //!< Error creating GDAL transformer
+      DestinationCreationError, //!< Error creating destination file
+      WarpFailure, //!< Failed warping source
     };
+    Q_ENUM( Result )
 
     /**
      * Warp the file specified by "input" and write the resulting raster to the file "output".
```

Comment:
```suggestion
     * Warp the file specified by \a input and write the resulting raster to the file \a output.
```

There are a few params missing (pre-existing compression and crs as well as newly added feedback), maybe worth adding those.